lang/wasm_alien_thread_runner.hh

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2023-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+
+#include <functional>
+#include <mutex>
+#include <queue>
+#include <condition_variable>
+#include <thread>
+
+#include <seastar/core/future.hh>
+
+#include "rust/cxx.h"
+#include "rust/wasmtime_bindings.hh"
+
+namespace wasm {
+
+struct wasm_compile_task {
+    std::function<void()> func;
+    seastar::promise<rust::Box<wasmtime::Module>>& done;
+    int shard;
```

Comment:
might want to use `unsigned`. to be consistent with the function signature of `seastar::alien::run_on()` . FWIW, we have `using shard_id = unsigned;`, and use `unsigned` and `shard_id` interchangeably.   but, in general, we don't use a signed int for shard.

Reply:
Makes sense